apps/dot_ext/forms.py

```diff
@@ -142,7 +145,77 @@ def save(self, *args, **kwargs):
                                                              app.name)
         logger.info(logmsg)
         app = super().save(*args, **kwargs)
-        uri = store_logo_image(self.cleaned_data.pop('logo_image', None), app.pk)
+        uri = app.store_media_file(self.cleaned_data.pop('logo_image', None), "logo.jpg")
+        if uri:
+            app.logo_uri = uri
+            app.save()
+        return app
+
+
+class CustomAdminApplicationForm(forms.ModelForm):
+
+    logo_image_bypass = forms.BooleanField(label='Bypass Logo Image Upload Restrictions', required=False,
+                                           help_text="If checked, this will bypass image type, size and dimension checks. ")
+    logo_image = logo_image_field
```

Comment:
If you want to extend the form with custom behavior but are worried about code reuse use inheritance.

Reply:
I like this idea since there is some duplicate code. TY!